x/bank/keeper/hooks_test.go

```diff
@@ -31,6 +31,21 @@ func (h *MockBankHooksReceiver) BeforeSend(ctx sdk.Context, from, to sdk.AccAddr
 	return nil
 }
 
+// variable for counting `TrackBeforeSend`
+var (
+	countTrackBeforeSend = 0
+	expNextCount         = 1
+)
+
+// Mock TrackBeforeSend bank hook that doesn't allow the sending of exactly 50 coins of any denom.
+func (h *MockBankHooksReceiver) TrackBeforeSend(ctx sdk.Context, from, to sdk.AccAddress, amount sdk.Coins) {
+	for _, coin := range amount {
+		if coin.Amount.Equal(sdk.NewInt(50)) {
+			countTrackBeforeSend += 1
+		}
+	}
+}
```

Comment:
I probably have a fundamental misunderstanding of how this is supposed to work, but isn't the whole point of TrackBeforeSend to simply track that a condition has been met but not stop the send? The godoc here and following tests imply it acts as both a block and tracker.